stable_baselines/common/policies.py

```diff
@@ -751,3 +786,23 @@ def register_policy(name, policy):
         raise ValueError("Error: the name {} is alreay registered for a different policy, will not override."
                          .format(name))
     _policy_registry[sub_class][name] = policy
+
+
+def create_dummy_action_mask(ac_space, num_samples):
+    if isinstance(ac_space, spaces.MultiDiscrete):
+        action_mask = np.ones((num_samples, ac_space.nvec[0], ac_space.nvec[1]), dtype=np.float32)
+    elif isinstance(ac_space, spaces.Discrete) or isinstance(ac_space, spaces.MultiBinary):
+        action_mask = np.ones((num_samples, ac_space.n), dtype=np.float32)
+    else:
+        action_mask = np.ones((num_samples, ac_space.shape[0]), dtype=np.float32)
+    return action_mask
+
+
+def reshape_action_mask(action_mask, ac_space, num_samples):
+    if isinstance(ac_space, spaces.MultiDiscrete):
+        action_mask = np.reshape(action_mask, (num_samples, ac_space.nvec[0], ac_space.nvec[1]), dtype=np.float32)
+    elif isinstance(ac_space, spaces.Discrete) or isinstance(ac_space, spaces.MultiBinary):
+        action_mask = np.reshape(action_mask, (num_samples, ac_space.n), dtype=np.float32)
+    elif isinstance(ac_space, spaces.Box):
```

Comment:
does masking make sense for continuous actions? or is it only to avoid errors?

Reply:
>Say the agent is controlling an n-dimensional robotic arm

Setting the action to zero does not prevent the arm from moving... (if the output is the position of the arm)

>it is to prevent errors as the step methods now require action_mask to be a variable in the graph

Maybe this can be solve using a placeholder with default value (so you don't have to feed the graph with a value for that tensor)